sdk/openai/Azure.AI.OpenAI/tests/AssistantTests.cs

```diff
@@ -455,13 +455,13 @@ public async Task FunctionToolsWork()
         Validate(run);
         Console.WriteLine($" Run status right after creation: {run.Status}");
 
-        // TODO FIXME: The underlying OpenAI code doesn't consider the "requires_action" status to be terminal even though it is
-        //             work around this here
-        for (int i = 0; i < 10 && !run.Status.IsTerminal && !run.Status.Equals("requires_action"); i++)
-        {
-            Thread.Sleep(500);
-            run = await client.GetRunAsync(run);
-        }
+        // TODO FIXME: The underlying OpenAI code doesn't consider the "requires_action" status to be terminal even though it is.
```

Comment:
This is expected (requires_action isn't terminal; the run is "paused" and not "stopped") but it reveals we have a bug in the other test code -- the samples fortunately get it right (requires_action check inside of the polling loop).

Reply:
I can understand that nuance. I guess it didn't match up with my expecations!